### PR TITLE
WHO 5 cards with Villainous Choice

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/e/ensnared_by_the_mara.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+SVar:DBDig:DB$ DigUntil | Defined$ Remembered | Valid$ Card.nonLand | FoundDestination$ Exile | RevealedDestination$ Exile | RememberFound$ True | SubAbility$ DBPlay | SpellDescription$ They exile cards from the top of their library until they exile a nonland card, then you may cast that card without paying its mana cost.
+SVar:DBPlay:DB$ Play | Controller$ You | Defined$ Remembered | WithoutManaCost$ True | ValidSA$ Spell | Optional$ True | Amount$ All | SubAbility$ DBCleanup
+SVar:DBDamage:DB$ Dig | RememberChanged$ True | DestinationZone$ Exile | Defined$ You |  DigNum$ 4 | ChangeNum$ All | SubAbility$ DamageOpponent | SpellDescription$ That player exiles the top four cards of their library and CARDNAME deals damage equal to the total mana value of those exiled cards to that player.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/e/ensnared_by_the_mara.txt:7]: wrong Defined?

File: forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Name:Genesis of the Daleks
+ManaCost:4 B B
+Types:Enchantment Saga
+K:Chapter:4:DBToken,DBToken,DBToken,DBVillainous
+SVar:DBToken:DB$ Token | TokenAmount$ X | TokenScript$ b_3_3_a_dalek_menace | SpellDescription$ Create a 3/3 black Dalek artifact creature token with menace for each lore counter on CARDNAME.
+SVar:DBVillainous:DB$ VillainousChoice | Choices$ DBDestroyDalek,DBDestroyNonDalek | ValidTgts$ Opponent | SpellDescription$ Target opponent faces a villainous choice — Destroy all Dalek creatures and each of your opponents loses life equal to the total power of Daleks that died this turn, or destroy all non-Dalek creatures.
+SVar:DBDestroyDalek:DB$ DestroyAll | ValidCards$ Creature.Dalek | RememberDestroyed$ True | SubAbility$ DBLoseLife | SpellDescription$ Destroy all Dalek creatures and each of your opponents loses life equal to the total power of Daleks that died this turn.
+SVar:DBLoseLife:DB$ LoseLife | LifeAmount$ TotalPower | Defined$ Opponent | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt:8]: wrong amount

File: forge-gui/res/cardsfolder/h/hunted_by_the_family.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Hunted by The Family
+ManaCost:5 U U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouDontCtrl | TgtPrompt$ Select up to four target creatures you don't control | TargetMin$ 0 | TargetMax$ 4 | RememberTargets$ True | SubAbility$ DBRepeat | SpellDescription$ Choose up to four target creatures you don't control. For each of them, that creature's controller faces a villainous choice — That creature becomes a 1/1 white Human creature and loses all abilities, or you create a token that's a copy of it.
+SVar:DBRepeat:DB$ RepeatEach | RepeatSubAbility$ DBChoice | RepeatCards$ Card.IsRemembered
+SVar:DBChoice:DB$ VillainousChoice | Defined$ RememberedController | Choices$ DBAnimate,DBCopy | SubAbility$ DBCleanup
+SVar:DBAnimate:DB$ Animate | Defined$ Remembered | Power$ 1 | Toughness$ 1 | Types$ Human | RemoveCreatureTypes$ True | RemoveAllAbilities$ True | IsCurse$ True | Colors$ White | OverwriteColors$ True | Duration$ Permanent | SpellDescription$ That creature becomes a 1/1 white Human creature and loses all abilities.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/h/hunted_by_the_family.txt:7]: all other types should be removed too

File: forge-gui/res/cardsfolder/h/hunted_by_the_family.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Name:Hunted by The Family
+ManaCost:5 U U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouDontCtrl | TgtPrompt$ Select up to four target creatures you don't control | TargetMin$ 0 | TargetMax$ 4 | RememberTargets$ True | SubAbility$ DBRepeat | SpellDescription$ Choose up to four target creatures you don't control. For each of them, that creature's controller faces a villainous choice — That creature becomes a 1/1 white Human creature and loses all abilities, or you create a token that's a copy of it.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/h/hunted_by_the_family.txt:4]: use ChooseCard so no Remember needed

File: forge-gui/res/cardsfolder/m/missy.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Name:Missy
+ManaCost:3 U B R
+Types:Legendary Creature Time Lord Rogue
+PT:4/5
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature.nonArtifact+Other+YouCtrl | TriggerZones$ Battlefield | Execute$ TrigChangeZone | TriggerDescription$ Whenever another nonartifact creature dies, return it to the battlefield under your control face down and tapped. It's a 2/2 Cyberman artifact creature.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/m/missy.txt:5]: wrong valid

**Review by tool4ever — CHANGES_REQUESTED** [review]

File: forge-gui/res/cardsfolder/h/hunted_by_the_family.txt (hunk 4 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/h/hunted_by_the_family.txt]: you still don't need Cleanup

File: forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt]: Well the approach to get the LKI is there now
but the power part is missing

Comment (Simisays) [reply forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt]: Hmm` Count$SumPower_ThisTurnEntered_Graveyard_from_Battlefield_Creature.Dalek` isn't working. Any pointers ?

File: forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+SVar:TrigVillainousChoice:DB$ VillainousChoice | Defined$ TriggeredDefendingPlayer | Choices$ DBSacrifice,DBGainControl
+SVar:DBSacrifice:DB$ Sacrifice | Amount$ 1 | SacValid$ Creature | Defined$ Remembered | SpellDescription$ That player sacrifices a creature.
+SVar:DBGainControl:DB$ GainControl | Defined$ ChosenCard | Choices$ Creature.RememberedPlayerCtrl | ChoiceTitle$ Select a creature that player controls | NewController$ You | LoseControl$ EOT | SubAbility$ DBTap | SpellDescription$ You gain control of a creature of your choice that player controls until end of turn. If you gain control of a creature this way, tap it, and it's attacking that player.
+SVar:DBTap:DB$ Tap | Defined$ ChosenCard | ConditionDefined$ ChosenCard | ConditionPresent$ Card.YouCtrl | SubAbility$ DBSetAttacking 
+Attacking$ RememberedPlayer |
```
Comment (Simisays) [inline forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt:10]: @Hanmac Any idea how this should be scripted ?

File: forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt]: We might need to add support if it's new

File: forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt (hunk 6 above)
Comment (Hanmac) [reply forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt:10]: @tool4ever i think this also needs extra support to make another card Attacking

File: forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt [hunk 7]
```diff
@@ -0,0 +1,14 @@
+Name:Midnight Crusader Shuttle
+ManaCost:4
+Types:Artifact Vehicle
+PT:3/4
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigVillainousChoice | TriggerDescription$ Midnight Entity — Whenever CARDNAME attacks, defending player faces a villainous choice — That player sacrifices a creature, or you gain control of a creature of your choice that player controls until end of turn. If you gain control of a creature this way, tap it, and it's attacking that player.
+SVar:TrigVillainousChoice:DB$ VillainousChoice | Defined$ TriggeredDefendingPlayer | Choices$ DBSacrifice,DBGainControl
+SVar:DBSacrifice:DB$ Sacrifice | Amount$ 1 | SacValid$ Creature | Defined$ Remembered | SpellDescription$ That player sacrifices a creature.
+SVar:DBGainControl:DB$ GainControl | Defined$ ChosenCard | Choices$ Creature.RememberedPlayerCtrl | ChoiceTitle$ Select a creature that player controls | NewController$ You | LoseControl$ EOT | SubAbility$ DBTap | SpellDescription$ You gain control of a creature of your choice that player controls until end of turn. If you gain control of a creature this way, tap it, and it's attacking that player.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt:8]: @tool4ever i think the GainControl Effect needs to use a Remember parameter to say that it should only remember the card if the controller changed (some say you can't change control)
and the following function should use RememberedCard

File: forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt (hunk 2 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt]: How should it be scripted after your changes ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/g/genesis_of_the_daleks.txt]: You should be able to chain it cleanly with handlePaid syntax now:
`Count$ThisTurnEntered_Graveyard_from_Battlefield_Dalek$CardPower`

File: forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt (hunk 6 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt:10]: will try to take a look soon :)

Comment (tool4ever) [reply forge-gui/res/cardsfolder/m/midnight_crusader_shuttle.txt:10]: @Simisays 
you can script it like the most recent Tahngarth version now

**Review by tool4ever — APPROVED** [review]

**Pull request merged**